### PR TITLE
expand examples

### PR DESCRIPTION
**Pull request by gracewilcox**
Adding onto the existing Monitor Query Readme to include more in-depth examples and context

### CODE REVIEW

File: sdk/monitor/azquery/README.md [hunk 1]
```diff
@@ -102,6 +102,9 @@ The timespan can be the following string formats:
 <duration> such as "P1Y2M10DT2H30M" // 1 year, 2 months, 10 days, 2 hours, 20 minutes
 ```
 
+### Logs Workspace IDs
```
Comment (jhendrixMSFT) [inline sdk/monitor/azquery/README.md:105]: Is this section supposed to be empty?

File: sdk/monitor/azquery/README.md [hunk 2]
```diff
@@ -117,66 +120,104 @@ The timespan can be the following string formats:
   - [Metrics result structure](#metrics-result-structure)
 
 ### Logs query
+The example below shows a basic logs query using the `QueryWorkspace` method. `QueryWorkspace` takes in a [context][context], a [Log Analytics Workspace][log_analytics_workspace] ID string, a [Body](#logs-query-body-structure) struct, and a [LogsClientQueryWorkspaceOptions](#increase-wait-time-include-statistics-include-render-visualization) struct and returns a [Results](#logs-query-result-structure) struct.
+
 ```go
-client := azquery.NewLogsClient(cred, nil)
-timespan := "2022-08-30/2022-08-31"
+import (
```
Comment (jhendrixMSFT) [inline sdk/monitor/azquery/README.md:126]: For full, end-to-end examples, we're moving away from putting them in markdown as they don't compile and it's too easy to end up with various copy/paste mistakes.

Take a look at some of the migration examples in [`azeventhubs`](https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/messaging/azeventhubs) and how they link to example files.

File: sdk/monitor/azquery/README.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/README.md:105]: It's not. Thanks for catching! Will fix.

Comment (gracewilcox) [reply sdk/monitor/azquery/README.md]: fixed in commit [d5e54b6](https://github.com/Azure/azure-sdk-for-go/pull/19208/commits/d5e54b6debc04da260e9cf6b8205c4565b57c56a)

File: sdk/monitor/azquery/README.md (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/README.md]: fixed in commit [d5e54b6](https://github.com/Azure/azure-sdk-for-go/pull/19208/commits/d5e54b6debc04da260e9cf6b8205c4565b57c56a)

File: sdk/monitor/azquery/example_test.go [hunk 3]
```diff
@@ -40,20 +41,38 @@ func ExampleNewMetricsClient() {
 }
 
 func ExampleLogsClient_QueryWorkspace() {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
```
Comment (jhendrixMSFT) [inline sdk/monitor/azquery/example_test.go:46]: I believe we've been using `log.Fatal()` in the examples.  Personally, I prefer `panic()` as it provides a stack trace so you can see what operation failed.  Take a look at some other examples and do what they do.

Comment (gracewilcox) [reply sdk/monitor/azquery/example_test.go]: using the TODO error handling format

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**